C/services/south/ingest.cpp

```diff
@@ -1002,11 +1008,18 @@ void Ingest::useFilteredData(OUTPUT_HANDLE *outHandle,
 			     READINGSET *readingSet)
 {
 	Ingest* ingest = (Ingest *)outHandle;
+
 	if (ingest->m_data != readingSet->getAllReadingsPtr())
 	{
-		if (ingest->m_data)
+		if (ingest->m_data && ingest->m_data->size())
 		{
-			ingest->m_data->clear();// Remove any pointers still in the vector
+			// Remove the readings in the vector
```

Comment:
I'm not sure this will work in all cases. I suspect we might be deleting the readings twice in C++ filters

Reply:
@MarkRiddoch : I tested again with sinusoid south service + asset filter (python) + metadata (C++) and scale (C++) filters, under valgrind, kept changing the filter order, I don't see any double free errors.

Reply:
That's good, thanks